File: src/VBWebView2_AddToCart/Form1.vb
Public Class Form1
    Sub New()
        InitializeComponent()
        Me.textBox1.Text = My.Resources.Url
    End Sub

    Async Sub InitializeAsync()
        Await Me.WebView2.EnsureCoreWebView2Async(Nothing)
    End Sub

    Private Sub Form1_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        Me.InitializeAsync()
    End Sub

    Private Sub button1_Click(sender As Object, e As EventArgs) Handles button1.Click
        Me.textBox1.Enabled = False
        Me.button1.Enabled = False
        Me.webView2.Source = New Uri(Me.textBox1.Text)
    End Sub

    Private Sub webView2_NavigationCompleted(sender As Object, e As Microsoft.Web.WebView2.Core.CoreWebView2NavigationCompletedEventArgs) Handles webView2.NavigationCompleted
        Dim source As String = Me.webView2.Source.ToString().ToLower()
        If source = Me.textBox1.Text.ToLower() Then
            Me.webView2.ExecuteScriptAsync(My.Resources.AddToCart)
        End If
    End Sub
End Class

File: src/CSWebView2_AddToCart/Form1.cs
using System.Windows.Forms;

namespace CSWebView2_AddToCart
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
            this.textBox1.Text = Properties.Resources.Url;
        }

        async void InitializeAsync()
        {
            await this.webView2.EnsureCoreWebView2Async(null);
        }

        private void Form1_Load(object sender, System.EventArgs e)
        {
            this.InitializeAsync();
        }

        private void button1_Click(object sender, System.EventArgs e)
        {
            this.textBox1.Enabled = false;
            this.button1.Enabled = false;
            this.webView2.Source = new System.Uri(this.textBox1.Text);
        }

        private void webView2_NavigationCompleted(object sender, Microsoft.Web.WebView2.Core.CoreWebView2NavigationCompletedEventArgs e)
        {
            var source = this.webView2.Source.ToString().ToLower();
            if (source == this.textBox1.Text.ToLower())
            {
                this.webView2.ExecuteScriptAsync(Properties.Resources.AddToCart);
            }
        }
    }
}
